jdatetime/__init__.py

```diff
@@ -900,58 +900,26 @@ def strptime(date_string, format):
                 "time data '%s' does not match format '%s'" %
                 (date_string, format))
 
-    def replace(
-            self,
-            year=None,
-            month=None,
-            day=None,
-            hour=None,
-            minute=None,
-            second=None,
-            microsecond=None,
-            tzinfo=None):
+    def replace(self, *args, **kwargs):
         """Return datetime with new specified fields."""
-        t_year = self.year
-        if year is not None:
-            t_year = year
-
-        t_month = self.month
-        if month is not None:
-            t_month = month
-
-        t_day = self.day
-        if day is not None:
-            t_day = day
-
-        t_hour = self.hour
-        if hour is not None:
-            t_hour = hour
-
-        t_min = self.minute
-        if minute is not None:
-            t_min = minute
-
-        t_sec = self.second
-        if second is not None:
-            t_sec = second
+        t_new = {}
+        for attr in ["year", "month", "day", "hour", "minute", "second",
```

Comment:
Here the PR tries to compact the list of parameters of the `replace` function, making it a shorter definition. Since the param list is getting long, it's understandable. However only using variadic parameters (`*args`) means that the function interface is not known to the caller anymore. So [inspect](https://docs.python.org/3/library/inspect.html) or [`replace.__code__.co_varnames`](https://python-reference.readthedocs.io/en/latest/docs/code/varnames.html#co-varnames) won't return the actual function parameters anymore, and editors won't be able to autocomplete when calling the function.

Maybe we can document the params list (with a function doc), I'm also OK with having a long function param list.
It's self documenting, and not that unusual in Python libraries. For example some famous projects keep long function parameters, like [pika](https://github.com/pika/pika/blob/160caa4e4621711290a915eec5f4a34c01ea2d1b/pika/spec.py#L696), [boto](https://github.com/boto/boto/blob/eca5f987aa59ec7490aa91466853c36b0cf40b3c/boto/pyami/installers/__init__.py#L30), [ansible](https://github.com/ansible/ansible/blob/40e5d2c951d5bc35d98fc47f66557713789f65c1/lib/ansible/plugins/callback/__init__.py#L380), [pycparser](https://github.com/eliben/pycparser/blob/2a29d56c280e64b308cd079b507ebee3e108fddc/pycparser/c_ast.py#L383).